model/generate.go

```diff
@@ -168,14 +172,19 @@ func (g *GeneratedProject) Save(ctx context.Context, p *Project, pp *ParserProje
 	return nil
 }
 
-// update the parser project using the newest config number (if using legacy version config, this comes from version)
-func updateParserProject(v *Version, pp *ParserProject) error {
+// updateParserProject updates the parser project along with generated task ID and updated config number
+// (if using legacy version config, this comes from version).
+func updateParserProject(v *Version, pp *ParserProject, taskId string) error {
+	if utility.StringSliceContains(pp.UpdatedByGenerators, taskId) {
+		// This generator has already updated the parser project so continue.
+		return nil
+	}
 	updateNum := pp.ConfigUpdateNumber + 1
-	// legacy: most likely a version for which no parser project exists
+	// Legacy: most likely a version for which no parser project exists.
 	if pp.ConfigUpdateNumber < v.ConfigUpdateNumber {
 		updateNum = v.ConfigUpdateNumber + 1
 	}
-
+	pp.UpdatedByGenerators = append(pp.UpdatedByGenerators, taskId)
```

Comment:
Why not do this after line 190 so that it doesn't set it if it errors?

Reply:
This needs to be upserted along with the rest of the parser project at 188